File: lib/exchange/models/publisher.dart
/// Mirrors `oep_exchange`'s `PublisherDto` wire shape (`GET /publishers`,
/// `GET /publishers/{id}`, `packages/api-contracts/src/publisher.ts`) —
/// field names are the wire format's own `camelCase` names, read
/// directly, matching `OfficialSource`'s own one-for-one mirroring of
/// EAM's wire shape.
class Publisher {
  const Publisher({
    required this.id,
    required this.namespace,
    required this.publisherType,
    required this.displayName,
    required this.legalName,
    required this.description,
    required this.website,
    required this.contactEmail,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
  });

  final String id;
  final String namespace;
  final String publisherType;
  final String displayName;
  final String legalName;
  final String description;
  final String website;
  final String contactEmail;
  final String status;
  final String createdAt;
  final String updatedAt;

  factory Publisher.fromJson(Map<String, Object?> json) => Publisher(
        id: json['id'] as String? ?? '',
        namespace: json['namespace'] as String? ?? '',
        publisherType: json['publisherType'] as String? ?? '',
        displayName: json['displayName'] as String? ?? '',
        legalName: json['legalName'] as String? ?? '',
        description: json['description'] as String? ?? '',
        website: json['website'] as String? ?? '',
        contactEmail: json['contactEmail'] as String? ?? '',
        status: json['status'] as String? ?? '',
        createdAt: json['createdAt'] as String? ?? '',
        updatedAt: json['updatedAt'] as String? ?? '',
      );
}

File: services/exchange/packages/api-contracts/src/publisher.ts
/**
 * Publisher wire contracts (TASK-EXC-0003, docs/tasks/WP-EXC-003.md §5).
 * `legalName` is `docs/tasks/WP-EXC-003.md`'s "Legal Name" field, mapped
 * to the Publisher persistence layer's `name` column (EXC-002 §4's
 * "Publisher Name") — see `docs/architecture/REPOSITORY_STRUCTURE.md`
 * for why this is a mapping rather than a new column. `namespace` and
 * `publisherType` aren't named in WP-EXC-003.md §5's field list but are
 * required by the already-approved schema (TASK-EXC-0002, EXC-002 §5/§6)
 * and so remain required here too.
 */
export type PublisherType =
  | 'individual'
  | 'company'
  | 'oem'
  | 'educational_institution'
  | 'government'
  | 'standards_organization'
  | 'enterprise'
  | 'community_organization';

export type PublisherStatus = 'active' | 'suspended';

export interface PublisherDto {
  id: string;
  namespace: string;
  publisherType: PublisherType;
  displayName: string;
  legalName: string;
  description: string;
  website: string;
  contactEmail: string;
  status: PublisherStatus;
  createdAt: string;
  updatedAt: string;
}

export interface CreatePublisherRequest {
  namespace: string;
  publisherType: PublisherType;
  displayName: string;
  legalName: string;
  contactEmail: string;
  description?: string;
  website?: string;
}

/** Fields a Publisher may change after creation — see `PublisherUpdate` in the persistence layer for why `namespace`/`publisherType`/`legalName` aren't included. */
export interface UpdatePublisherRequest {
  displayName?: string;
  contactEmail?: string;
  description?: string;
  website?: string;
  status?: PublisherStatus;
}

export interface PublisherListResponse {
  publishers: PublisherDto[];
}
